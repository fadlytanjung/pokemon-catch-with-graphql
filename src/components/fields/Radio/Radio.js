import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Radio(props) {
  const { data, input } = props;
  const labelChecked = [styles.root, styles['label-checked']].filter(Boolean).join(' ');

  return (
    <label className={data.value === input.value ? labelChecked : styles.root}>
      {data.text}
      <input {...input} type="radio" value={data.value} />
    </label>
  );
}

Radio.defaultProps = {
  className: '',
  data: {},
  input: {},
};

Radio.propTypes = {
  className: PropTypes.string,
  data: PropTypes.shape({
    text: PropTypes.string,
    value: PropTypes.string,
  }),
  input: PropTypes.object,
};
