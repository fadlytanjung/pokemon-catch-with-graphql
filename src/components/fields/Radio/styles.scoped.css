.root {
  border: 0.0625rem solid #cccccc;
  border-radius: 0.5rem;
  padding: 0.75rem;
  text-align: center;
  width: 100%;
}

.root:hover {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  transition: all 0.35s ease-in-out;
}

.root > input[type="radio"] {
  opacity: 0;
  position: fixed;
  width: 0;
}

.root.label-checked {
  border: 0.0625rem solid #0eb37e;
  background-color: #0eb37e;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  color: #ffffff;
}
