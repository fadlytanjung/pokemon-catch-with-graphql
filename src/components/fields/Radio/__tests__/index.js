import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Radio from '../Radio';

jest.mock('../styles.scoped.css', () => ({
  root: 'root',
}));

describe('src/components/fields/Radio', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Radio />);
    expect(tree).toMatchSnapshot();
  });

  test('render unchecked', () => {
    const { defaultProps } = Radio;
    const data = { value: 'data' };
    const input = { value: 'input' };
    const result = Radio({ ...defaultProps, data, input });
    expect(result.props.className).toBe('root');
  });
});
