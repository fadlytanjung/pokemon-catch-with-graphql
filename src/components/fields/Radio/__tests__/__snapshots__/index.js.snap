// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Radio render 1`] = `
<label
  className="root"
>
  <input
    type="radio"
  />
</label>
`;
